modules/reindex/src/test/java/org/elasticsearch/reindex/ReindexBasicTests.java

```diff
@@ -37,21 +41,32 @@ public void testFiltering() throws Exception {
         );
         assertHitCount(prepareSearch("source").setSize(0), 4);
 
+        TestTelemetryPlugin testTelemetryPlugin = getTestTelemetryPlugin();
+        testTelemetryPlugin.resetMeter();
         // Copy all the docs
         ReindexRequestBuilder copy = reindex().source("source").destination("dest").refresh(true);
+        testTelemetryPlugin.collect();
```

Comment:
we're calling `collect()` before the reindex is executed, we should call that _after_

Reply:
Thanks! I wasn't quite sure how collect works.